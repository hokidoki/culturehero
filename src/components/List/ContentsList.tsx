import ContentsListStyles from './ContentsList.styles';
import { ContentsListProps } from './ContentsList.types';
import { ContentsListItem } from '../ListItem/ContentsListItem';

export default function ContentsList({ contents }: ContentsListProps) {
  return (
    <ContentsListStyles.Layout>
      <ContentsListStyles.AddContentButton />
      {contents.map((v) => (
        <ContentsListItem {...v} key={v.id} />
      ))}
    </ContentsListStyles.Layout>
  );
}
