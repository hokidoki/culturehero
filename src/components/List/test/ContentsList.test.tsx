import { fireEvent, render, waitFor } from '@render';
import ContentsList from '../ContentsList';
import { isNull } from '@fxts/core';

describe('ContentsList', () => {
  it('match snapshot', () => {
    const props = {
      contents: [
        {
          id: '1',
          summary: 'summary',
          title: 'title',
          createdAt: '2020-02-02',
        },
      ],
      newArticleButtonOnClick: () => {},
    };
    const utils = render(<ContentsList {...props} />);
    expect(utils.container).toMatchSnapshot();
  });

  it('리스트의 버튼을 클릭시 프롭스로 전달한 newArticleButtonOnClick이 호출 된다. ', () => {
    const props = {
      contents: [
        {
          id: '1',
          summary: 'summary',
          title: 'title',
          createdAt: '2020-02-02',
        },
      ],
      newArticleButtonOnClick: jest.fn(),
    };
    const utils = render(<ContentsList {...props} />);
    const button = utils.container.querySelector('button');

    if (isNull(button)) throw new Error('button missing');

    fireEvent.click(button);
    expect(props.newArticleButtonOnClick).toBeCalled();
  });
});
