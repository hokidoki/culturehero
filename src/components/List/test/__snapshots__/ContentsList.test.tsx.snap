// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentsList match snapshot 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  background-color: rgb(17,17,20);
  padding: 40px;
  gap: 20px;
  max-width: 1000px;
}

.c1 {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  border: 1px solid rgb(46,48,51);
  position: relative;
  border-radius: 15px;
  transition: background-color 0.3s;
  width: 100%;
  height: 100px;
  padding: 10px;
  background-color: transparent;
}

.c1::before,
.c1::after {
  content: "";
  border-radius: 2px;
  width: 5px;
  height: 40px;
  position: absolute;
  top: 50%;
  left: 50%;
  background-color: rgb(46,48,51);
  transform: translate(-50%,-50%);
}

.c1::before {
  width: 40px;
  height: 5px;
}

.c1:hover {
  background-color: rgb(33,33,36);
  cursor: pointer;
}

.c2 {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  border: 1px solid rgb(46,48,51);
  border-radius: 15px;
  transition: background-color 0.3s;
  width: 100%;
  height: 100px;
  padding: 10px;
}

.c2:hover {
  background-color: rgb(33,33,36);
  cursor: pointer;
}

.c3 {
  color: #9aa0a6;
  font-size: 1.25rem;
}

.c4 {
  color: #bdc1c6;
  font-size: 1.25rem;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
    />
    <a
      href="/contents/1"
      style="display: contents;"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          title
        </span>
        <p
          class="c4"
        >
          summary
        </p>
      </div>
    </a>
  </div>
</div>
`;
